views/js/pciCreator/dev/audioRecordingInteraction/runtime/audioRecordingInteraction.js

```diff
@@ -661,38 +671,50 @@ define([
          * @param {Object} interaction
          */
         destroy: function destroy() {
-            if (this.recorder && this.recorder.is('recording')) {
-                this.stopRecording();
-            }
-            if (this.player && this.player.is('playing')) {
-                this.stopPlayback();
-            }
+            var self = this;
 
-            this.destroyControls();
+            return new Promise(function(resolve, reject) {
+                var promises = [];
 
-            if (this.inputMeter) {
-                this.inputMeter.destroy();
-                this.inputMeter = null;
-            }
+                if (self.recorder && self.recorder.is('recording')) {
+                    promises.push(self.stopRecording());
+                }
+                if (self.player && self.player.is('playing')) {
+                    promises.push(self.stopPlayback());
+                }
 
-            this.progressBar = null;
+                promises.push(self.destroyControls());
 
-            if (this.mediaStimulus) {
-                this.mediaStimulus.destroy();
-                this.mediaStimulus = null;
-            }
+                if (self.inputMeter) {
+                    promises.push(self.inputMeter.destroy());
+                }
 
-            if (this.player) {
-                this.player.unload();
-                this.player = null;
-            }
+                if (self.mediaStimulus) {
+                    promises.push(self.mediaStimulus.destroy());
+                }
 
-            if (this.recorder) {
-                this.recorder.destroy();
-                this.recorder = null;
-            }
+                if (self.player) {
+                    promises.push(self.player.unload());
+                }
 
-            this.resetResponse();
+                if (self.recorder) {
+                    promises.push(self.recorder.destroy());
+                }
+
+                promises.push(self.resetResponse());
+
+                Promise.all(promises).then(
```

Comment:
Promises chains,  so you don't need to wrap this in a top level Promise : 

```js
destroy: function destroy() {
    var promises = [];
    //....
    return Promise
        .all(promises)
        .then(function(){
            //...
         });
}
```
And you don't even need to handle the cactch/reject clauses, it will be handled by the caller